Userland/Libraries/LibCore/System.h

```diff
@@ -178,7 +178,7 @@ enum class SearchInPath {
 ErrorOr<void> exec_command(Vector<StringView>& command, bool preserve_env);
 #endif
 
-ErrorOr<void> exec(StringView filename, Span<StringView> arguments, SearchInPath, Optional<Span<StringView>> environment = {});
+ErrorOr<void> exec(StringView filename, Span<StringView const> arguments, SearchInPath, Optional<Span<StringView const>> environment = {});
```

Comment:
I can see why this is needed, but it feels like there should be a better way to do this (note that I don't know if that better way actually exists).

Our `StringView` are by definition read-only and don't feature any non-const methods (as far as I can see), so every `StringView` is implicitly const, even if C++ doesn't recognize it as such. We should probably either find a way to make every StringView C++-const without qualifying it everywhere, or we just leave it as non-const and fix up the few places where we have defined it as const (by "accident"). Starting to mix `const` and non-`const` `StringView`s even more will probably create more work down the line.

Reply:
It's a little confusing but this constness is not referring to the constness of the StringView, but to the constness of the underlying array of StringViews. So that this is saying is that `exec()` won't try to mutate the argument span (i.e. by putting other StringViews in there).

Reply:
We probably should just add a `ReadonlySpan<T>` alias to make this clearer.

Reply:
Yeah, to be clear this is needed when the caller has a container such as:
```c++
Vector<StringView> const& some_parameter
```

Converting `some_parameter` to a `Span` (either implicitly or via `.span()` will only ever return a `Span<StringView const>` here, and will not compile if passed to a function accepting only `Span<StringView>`.